### PR TITLE
Edit general secret rotation info.

### LINKED ISSUE #743: [CAP-986] How to rotate a secret
https://github.com/cloudfoundry-incubator/kubecf/blob/master/doc/secret_rotation_general.md

### PR DESCRIPTION
**Pull request by KucharczykL**
Closes #743.

### CODE REVIEW

File: xml/cap_admin_certificates.xml [hunk 1]
```diff
@@ -209,40 +209,75 @@
...
+   From time to time, operators may wish to change such secrets, either manually
+   or on a schedule. This is called rotating a secret.
   </para>
+  <sect2 xml:id="sec-cap-secrets-rotation-finding">
+    <title>Finding secrets</title>
+    <para>
+     Retrieve the list of all secrets maintained by &kubecf;:
+    </para>
+<screen>
+&prompt.user;kubectl get quarkssecret --namespace kubecf
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: We should put the namespace in `replaceable` tags. Applies to other instances as well. (and future content we work on)

File: xml/cap_admin_certificates.xml [hunk 2]
```diff
@@ -209,40 +209,75 @@
...
-  # Increment this counter to rotate all generated secrets
-  secrets_generation_counter: 2</screen>
-
-  <para>
-   Next, perform a <command>helm upgrade</command> to apply the change.
+   From time to time, operators may wish to change such secrets, either manually
+   or on a schedule. This is called rotating a secret.
   </para>
+  <sect2 xml:id="sec-cap-secrets-rotation-finding">
+    <title>Finding secrets</title>
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: Needs to be title-style capitalization. Applies to other `title` elements too.

https://doc.opensuse.org/products/opensuse/Styleguide/opensuse_documentation_styleguide_sd/#sec.capitalization

File: xml/cap_admin_certificates.xml [hunk 3]
```diff
@@ -209,40 +209,75 @@
...
+     Note that each quarkssecret has a corresponding regular &kube; secret
+     that it controls:
+   </para>
+<screen>
+&prompt.user;kubectl get secret --namespace kubecf
+&prompt.user;kubectl get secret --namespace kubecf kubecf.var-nats-password --output yaml
+   </screen>
+  </sect2>
+  <sect2 xml:id="sec-cap-secrets-rotation-specific">
+    <title>Rotating specific secret</title>
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: Think this should be either "Rotating a Specific Secret" or "Rotating Specific Secrets"

File: xml/cap_admin_certificates.xml [hunk 4]
```diff
@@ -209,40 +209,75 @@
...
+    </para>
+    <procedure>
+      <step>
+        <para>Create a YAML file for a ConfigMap of the form:</para>
+<screen>
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rotate-kubecf.var-nats-password
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: `replaceable` could be used here too I guess?

File: xml/cap_admin_certificates.xml [hunk 5]
```diff
@@ -209,40 +209,75 @@
...
+  labels:
+    quarks.cloudfoundry.org/secret-rotation: "true"
+data:
+  secrets: '["kubecf.var-nats-password"]'
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: Would be good to include the notes from the reference. Especially the one about `data.secrets` accepting an array

File: xml/cap_admin_certificates.xml [hunk 6]
```diff
@@ -209,40 +209,75 @@
...
+<screen>
+&prompt.user;kubectl apply --namespace kubecf -f /path/to/your/yaml/file
+       </screen>
+       <para>
+         The result can be seen in the &operator;'s log. After the rotation
+         is complete, that is after secrets have been changed and all affected
+         pods have been restarted, delete the config map again:
+       </para>
+<screen>
+&prompt.user;kubectl delete --namespace kubecf -f /path/to/your/yaml/file
```
Comment (btat) [inline xml/cap_admin_certificates.xml]: Might be better to split the creation and deletion parts?

**Comment by KucharczykL** [comment]
Thank you for all the feedback, I will keep in mind in the future :)

**Pull request merged**